design/resource-dependencies.md

```diff
@@ -0,0 +1,118 @@
+# Resource dependency relationships
+
+### Inventory of resources that could have dependency relationships
+All Rook-Ceph CRDs:
+- CephCluster
+- CephBlockPool
+- CephFilesystem
+- CephObjectStore
+- CephNFS
+- CephClient
+- CephRBDMirror
+- CephFilesystemMirror
+- CephObjectRealm
+- CephObjectZoneGroup
+- CephObjectZone
+
+Resources for which Rook-Ceph acts as a driver/provisioner:
+- ObjectBucketClaim (lib-bucket-provisioner)
+- ObjectBucket (COSI)
+
+Kubernetes resources which can depend on Rook:
+- StorageClass
+- PersistentVolume
+
+## Proposed dependency relationships
+
+#### `CephCluster`
+Primary dependents which can create/delete pools:
```

Comment:
All Rook CRs in the same namespace would be dependents of the CephCluster CR, right? Any of them should block the CephCluster removal. (edit: I see a comment about this at the end of the doc, if it could just be clarified here)

Reply:
Right! And that seems a good clarification.